### PR TITLE
Feature/pass in all options through canned queries and hooks sections

### PR DESCRIPTION
**Pull request by madeline-scyphers**
## Description

- Rewrites some of the configuration inputs so that instead of the config filing taking a section called `"canned_queries"` and `"getters_and_setters"`, as well as having a `"command_readers"` section for other Command Readers or new/custom Command Readers, all Command Readers now are done the same way, under the `"command_readers"` section.
- Also did the same thing for hooks, so all hooks and command readers can be passed in just inside the `"hooks"` and `"command_readers"` section of the config.
- Also made the main Cereal __init__ just take the different sections of the config directly, instead of being passed in a dictionary, and parsing those. (i.e. `Cereal(command_readers=blah, hooks=blah, encoding=blah, etc)`)
- Created a subclass of `Cereal` called `MockSerial` that can take the exact configuration format of pre 0.9.0 GRANOLA, but will issue deprecation warnings and tell people to switch over to `Cereal` instead.

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [x]  🗑️ Deprecation (change to deprecate existing code or functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [x] I've written tests for all new methods and classes that I created.
- [x] I've written the docstring in `Google` format for all the methods and classes that I used.

### CODE REVIEW

File: docs/config/canned_queries.rst [hunk 1]
```diff
@@ -2,7 +2,8 @@
 Canned Queries Configuration
 =================================
 
-To use the :mod:`Command Reader <granola.command_readers>` :class:`~granola.command_readers.CannedQueries`, you must define ``"canned_queries"`` as a dictionary in your configuration.
+To use the :mod:`Command Reader <granola.command_readers>` :class:`~granola.command_readers.CannedQueries`,
+you must define ``"canned_queries"`` as a dictionary in your configuration.
```
Comment (prosaicpudding) [inline docs/config/canned_queries.rst:6]: canned_queries, or CannedQueries

**Review by prosaicpudding — APPROVED** [review]
Just a few minor clean-ups. This basically looks good to me.

File: granola/tests/test_setup.py [hunk 2]
```diff
@@ -200,3 +207,54 @@ def test_a_json_config_can_specify_command_readers_as_well():
...
+        def get_reading(self, data):
+            if "1" in data:
+                return "Command Reader1"
+
+    class CommandReader2(BaseCommandReaders):
+        def get_reading(self, data):
+            if "2" in data:
+                return "Command Reader2"
+
+    # and a config dictionary that reference the command readers by name and by the actual object
```
[the remark this answers is not in the document]
Comment (madeline-scyphers) [reply granola/tests/test_setup.py]: thanks for catching that. Updated

File: docs/config/canned_queries.rst (hunk 1 above)
Comment (madeline-scyphers) [reply docs/config/canned_queries.rst]: Thanks for catching that, updated.

**Pull request merged**